source/common/router/upstream_request.cc

```diff
@@ -72,6 +244,28 @@ UpstreamRequest::UpstreamRequest(RouterFilterInterface& parent,
   if (cluster_info.has_value()) {
     stream_info_.setUpstreamClusterInfo(*cluster_info);
   }
+  if (!allow_upstream_filters_) {
+    return;
+  }
+
+  // Set up the upstream filter manager.
+  fm_callbacks_ = std::make_unique<UpstreamRequestFilterManagerCallbacks>(*this);
```

Comment:
nit: I would call this filter_manager_callbacks_ to avoid abbreviations, but your call as that's pretty verbose.